fix: Fix MAv2 React Hook client creation for 7702

jakehobbs opened the pull request:
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

adamegyed commented on account-kit/core/src/actions/createAccount.ts:
```diff
@@ -45,9 +45,11 @@ export type AccountConfig<TAccount extends SupportedAccountTypes> =
       >
-    : OmitSignerTransportChain<
+    : TAccount extends "ModularAccountV2"
+    ? OmitSignerTransportChain<
         CreateModularAccountV2Params<Transport, AlchemyWebSigner>
-      >;
+      >
+    : never;
```
Just so I understand the old behavior: previously, it would fall into the MAv2 branch as a default branch, and now we're asserting that the user has to provide "ModularAccountV2" as a mode?

jakehobbs replied:
This change isn't necessary. I was just trying to make it more explicit.

adamegyed replied:
It makes sense! Just wanted to clarify

adamegyed commented on account-kit/core/src/actions/getSmartAccountClient.ts:
```diff
@@ -218,12 +220,23 @@ export function getSmartAccountClient(
           isLoadingClient: false,
         };
       case "ModularAccountV2":
+        const mode = (accountParams as AccountConfig<"ModularAccountV2">).mode;
```
If we avoid destructuring `accountParams` earlier, can we avoid the `as` casting here? It should propagate the type inference as being of `"ModularAccountV2:` thought the generic args.

jakehobbs replied:
been fighting this for a bit but cannot seem to make it infer properly

jakehobbs replied:
pushed a change to make it a bit safer at least

adamegyed reviewed (commented):
Looks good! Had a few small questions.

Also, since this PR is not touching code in the `examples/ui-demo/` folder, could you set the base as `adam/7702-core`? That way we can merge it in before demo liveness.

jakehobbs commented:
> Looks good! Had a few small questions.
> 
> Also, since this PR is not touching code in the `examples/ui-demo/` folder, could you set the base as `adam/7702-core`? That way we can merge it in before demo liveness.

i changed the base, also fixed an issue w/ MAv2 accounts not reconnecting after page refresh.

jakehobbs commented:
@adamegyed just fixed docs too fyi

adamegyed reviewed (approved):
Looks good!
